src/pages/Upload.js

```diff
@@ -757,7 +757,7 @@ const MetadataUploadPanel = ({
       h(
         Dropzone,
         {
-          disabled: !!WorkspaceUtils.editWorkspaceError(workspace),
+          disabled: !WorkspaceUtils.canEditWorkspace(workspace)[0],
```

Comment:
I agree with Nick that the signature is a nice improvement from before, but usage like this I think is a cryptic result of using tuple return types.... if instead we returned a small object with named properties, the main usage would only change slightly:

`const { canEdit, editErrorMessage } = canEditWorkspace(workspace);`

and this usage here would become much more readable:

`disabled: !WorkspaceUtils.canEditWorkspace(workspace).canEdit,`

Reply:
Indeed, I do not love this, but the anonymity of the variable names results in a clearer API than `canEditWorkspace.canEdit`, in my opinion, which is confusing. The more general question I am having to answer here is, "How do I write an either without Either?" In that case, Right and Left become semi-anonymous names that prevent the above confusion. If I do name them, I'd like to keep them generic, such as:
```javascript
const { value, message } = canEditWorkspace(workspace);
```

Reply:
https://github.com/DataBiosphere/terra-ui/pull/4268/commits/6e0aafcc1b53570cb7a15e44fbbee7f9279c5739